applications/FluidDynamicsApplication/custom_processes/distance_smoothing_process.h

```diff
@@ -70,32 +92,188 @@ class KRATOS_API(FLUID_DYNAMICS_APPLICATION) DistanceSmoothingProcess : public P
+            CreateAuxModelPart();
+        }
+
+        auto& r_smoothing_model_part = mrModel.GetModelPart( mAuxModelPartName );
+
+        block_for_each(r_smoothing_model_part.Nodes(), [&](Node<3>& rNode){
+                rNode.Free(DISTANCE);
+                const double distance = rNode.FastGetSolutionStepValue(DISTANCE);
+                rNode.FastGetSolutionStepValue(DISTANCE, 1) = distance;
+            });
+
+        mp_solving_strategy->Solve();
+
+        auto &r_communicator = r_smoothing_model_part.GetCommunicator();
+        r_communicator.SynchronizeCurrentDataToMin(DISTANCE);
+
+        block_for_each(r_smoothing_model_part.Nodes(), [&](Node<3>& rNode){
+                rNode.SetValue( DISTANCE, rNode.FastGetSolutionStepValue(DISTANCE)
+                    - rNode.FastGetSolutionStepValue(DISTANCE, 1) ); // Corrected distance difference
+            });
+
+        auto& r_data_comm = r_smoothing_model_part.GetCommunicator().GetDataCommunicator();
+        GlobalPointersVector< Node<3 > > gp_list;
+
+        for (int i_node = 0; i_node < static_cast<int>(r_smoothing_model_part.NumberOfNodes()); ++i_node){
+            auto it_node = r_smoothing_model_part.NodesBegin() + i_node;
+            auto& global_pointer_list = it_node->GetValue(NEIGHBOUR_NODES);
+
+            for (unsigned int j = 0; j< global_pointer_list.size(); ++j)
+            {
+                auto& global_pointer = global_pointer_list(j);
+                gp_list.push_back(global_pointer);
+            }
+        }
+
+        GlobalPointerCommunicator< Node<3 > > pointer_comm(r_data_comm, gp_list);
+
+        auto coordinate_proxy = pointer_comm.Apply(
+            [](GlobalPointer<Node<3> >& global_pointer) -> Point::CoordinatesArrayType
+            {
+                return global_pointer->Coordinates();
+            }
+        );
+
+        auto distance_proxy = pointer_comm.Apply(
+            [&](GlobalPointer<Node<3> >& global_pointer) -> double
+            {
+                return global_pointer->GetValue(DISTANCE);
+            }
+        );
+
+        auto contact_proxy = pointer_comm.Apply(
+            [&](GlobalPointer<Node<3> >& global_pointer) -> bool
+            {
+                return global_pointer->Is(CONTACT);
+            }
+        );
+
+        r_communicator.GetDataCommunicator().Barrier();
+
+        block_for_each(r_smoothing_model_part.Nodes(), [&](Node<3>& rNode){
+            const auto& x_i = rNode.Coordinates();
+
+            double weight = 0.0;
+            double dist_diff_avg = 0.0;
+
+            GlobalPointersVector< Node<3 > >& global_pointer_list = rNode.GetValue(NEIGHBOUR_NODES);
+
+            for (unsigned int j = 0; j< global_pointer_list.size(); ++j)
+            {
+                auto& global_pointer = global_pointer_list(j);
+                const auto x_j = coordinate_proxy.Get(global_pointer);
+
+                if (contact_proxy.Get(global_pointer) == rNode.Is(CONTACT)){
```

Comment:
```suggestion
                if (contact_proxy.Get(global_pointer) && rNode.Is(CONTACT)){
```
Not sure if this is the intended behavior, but the current condition will enter the if with both `true-true` and `false-false`.

Reply:
The original one (true-true or false-false) is correct. Nodes are separated to two categories (CONTACT and not CONTACT).